### ISSUE TITLE
Use conjugate priors in `Normal` to simplify hyperparameter inference and reduce the grid dimensionality

### ISSUE THREAD

**Issue by emilyfertig** [probcomp/hierarchical-irm#127]

**Comment by Schaechtle** [comment]
Naive question: isn't the [Normal](https://github.com/probcomp/hierarchical-irm/blob/master/cxx/distributions/normal.cc) already in conjugate, Gaussian inverse Gamma form?

**Comment by emilyfertig** [comment]
Srinivas had an idea for specifying the Normal model differently so that it's still conjugate, but we only have to infer 3 hyperparameters in `transition_hyperparameters` instead of 4. @srvasude could you summarize what it was?

**Comment by srvasude** [comment]
Currently the hyperprior isn't conjugate. I'm considering (will need to work out the math for this) of using a partially conjugate hyperprior to the prior parameters: the mean and variance for the mean prior and for the scale parameter of the variance prior. The shape parameter of the inverse gamma/gamma would still need to be a grid prior, so the thought would be maybe you only need to grid on that, and can sample exactly from everything else.
